packages/lib/gpc/src/gpcCompile.ts

```diff
@@ -392,7 +392,8 @@ function compileProofListMembership<
     return BigInt(idx);
   });
 
-  // Compile listValidValues
+  // Compile listValidValues, making sure to sort the hashed values before
```

Comment:
This is good, though it suggests that maybe binding/canonicalization should sort the values as well.  Or maybe we should change the public representation of membership lists to be a Set?
(No need to block this PR on this, but it's something to consider.)